.github/workflows/pr-test.yaml

```diff
@@ -1,11 +1,10 @@
+name: "Test PR"
+
 on:
-  push:
-    branches:
-      - master
```

Comment:
I would keep this for the sole reason that even with requiring branches to be up to date, things can slip through. CI minutes are free, so it's better to have this check than not.